src/main/java/org/jenkinsci/plugins/mesos/MesosRetentionStrategy.java

```diff
@@ -94,6 +96,22 @@ private long checkInternal(MesosComputer c) {
         }
       }
     }
+
+    // Terminate the computer if it is exists for longer than
+    // 'maximumTimeToLive'.
+    final long timeLivedInMilliseconds =
+            DateTimeUtils.currentTimeMillis() - c.getConnectTime();
+
+    if (c.isOnline()) {
+
+      if (timeLivedInMilliseconds > MINUTES.toMillis(maximumTimeToLive)) {
+        LOGGER.info("Disconnecting computer greater maximum TTL " + c.getName());
+
```

Comment:
I don't think there is something better than setting a pending delete. This
will give jenkins the time to clean up.
Am 14.06.2016 03:27 schrieb "Ryoichi Taniguchi" <notifications@github.com>:

> In src/main/java/org/jenkinsci/plugins/mesos/MesosRetentionStrategy.java
> <https://github.com/jenkinsci/mesos-plugin/pull/173#discussion_r66896630>:
>
> > @@ -94,6 +96,22 @@ private long checkInternal(MesosComputer c) {
> >          }
> >        }
> >      }
> > +
> > +    // Terminate the computer if it is exists for longer than
> > +    // 'maximumTimeToLive'.
> > +    final long timeLivedInMilliseconds =
> > +            DateTimeUtils.currentTimeMillis() - c.getConnectTime();
> > +
> > +    if (c.isOnline()) {
> > +
> > +      if (timeLivedInMilliseconds > MINUTES.toMillis(maximumTimeToLive)) {
> > +        LOGGER.info("Disconnecting computer greater maximum TTL " + c.getName());
> > +
>
> Let me ask a question. Can slave be removed without setting pendingDelete,
> is anything else of safe clener here except for MesosCleanupThread ?
> (I look into these piece of codes recently, suffering from slave
> connectivety issue between plugin and mesos/zookeeper in my test
> environment.)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/mesos-plugin/pull/173/files/b448cf66877689cf259550aa7adc112a280341c6#r66896630>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADm-eiCrcISJ4Iztwmu1N9sDam7ykEzqks5qLgNzgaJpZM4GEdUV>
> .
>